tests/benchmark/utils/s2n_negotiate_client_benchmark.cc

```diff
@@ -0,0 +1,229 @@
+            &cipher_preferences_benchmark, /* cipher_preferences */
+            &kem_preferences_kms_pq_tls_1_0_2020_07, /* kem_preferences */
+            &s2n_signature_preferences_20201021, /* signature_preferences */
+            NULL, /* certificate_signature_preferences */
+            &s2n_ecc_preferences_20201021, /* ecc_preferences */
+    };
+
+    config->security_policy = &security_policy_benchmark;
+
+    struct verify_data *unsafe_verify_data = (verify_data *) malloc(sizeof(verify_data));
+    s2n_status_request_type type = S2N_STATUS_REQUEST_NONE;
+
+    GUARD_EXIT(s2n_config_set_status_request_type(config, type),
+               "OCSP validation is not supported by the linked libCrypto implementation. It cannot be set.");
+
+    if (s2n_config_set_verify_host_callback(config, unsafe_verify_host, unsafe_verify_data) != S2N_SUCCESS) {
+        print_s2n_error("Error setting host name verification function.");
+    }
+
+    unsafe_verify_data->trusted_host = host;
+
+    if (!conn_settings.insecure) {
+        const char *ca_dir = NULL;
+        if(all_suites[state.range(0)]->auth_method == S2N_AUTHENTICATION_RSA) {
+            std::string ca_file = "rsa_2048_sha384_client_cert.pem";
+            std::string pem_file_location = pem_dir + ca_file;
+            if (s2n_config_set_verification_ca_location(config, pem_file_location.c_str(), ca_dir) < 0) {
+                print_s2n_error("Error setting CA file for trust store.");
+            }
+        }
+        else {
+            std::string ca_file = "ecdsa_p256_pkcs1_cert.pem";
+            std::string pem_file_location = pem_dir + ca_file;
+            if (s2n_config_set_verification_ca_location(config, pem_file_location.c_str(), ca_dir) < 0) {
+                print_s2n_error("Error setting CA file for trust store.");
+            }
+        }
+    }
+    else {
+        GUARD_EXIT(s2n_config_disable_x509_verification(config), "Error disabling X.509 validation");
+    }
+
+    if (conn == NULL) {
+        print_s2n_error("Error getting new connection");
+        exit(1);
+    }
+
+    GUARD_EXIT(s2n_connection_set_config(conn, config), "Error setting configuration");
+
+    GUARD_EXIT(s2n_connection_set_client_auth_type(conn, S2N_CERT_AUTH_OPTIONAL),
+               "Error setting ClientAuth optional");
+
+    if (conn_settings.use_corked_io) {
+        GUARD_EXIT(s2n_connection_use_corked_io(conn), "Error setting corked io");
+    }
+
+    for (size_t i = 0; i < WARMUP_ITERS; i++) {
+        client_handshake(state, true, conn);
+    }
+    for (auto _ : state) {
+        state.PauseTiming();
+        client_handshake(state, false, conn);
+    }
+    free(unsafe_verify_data);
+    GUARD_EXIT(s2n_config_free(config), "Error freeing configuration");
+    GUARD_EXIT(s2n_connection_free(conn), "Error freeing connection");
+}
+
+int start_negotiate_benchmark_client(int argc, char** argv) {
+    int use_corked_io = 0;
+    int insecure = 1;
```

Comment:
This should default to benchmarking with certificate validation. Customers might not realize they are skipping the validation time when running with all default arguments.